docs/docs/portals.md

```diff
@@ -0,0 +1,89 @@
+---
+id: portals
+title: Portals
+permalink: docs/portals.html
+---
+
+Portals provide a first-class way to render children into a DOM node that exists outside the DOM hierarchy of the parent component.
+
+```js
+ReactDOM.createPortal(
+  child,
+  container
+);
+```
+
+The first argument (`child`) is any [renderable React child](/docs/react-component.html#render), such as an element, string, or fragment. The second argument (`container`) is a DOM element.
+
+## Usage
+
+Normally, when you return an element from a component's render method, it's mounted into the DOM as a child of the nearest parent node:
+
+```js{4,6}
+render() {
+  // React mounts a new div and renders the children into it
+  return (
+    <div>
+      {this.props.children}
+    </div>
+  );
+}
+```
+
+However, sometimes it's useful to insert a child into a different location in the DOM:
+
+```js{6}
+render() {
+  // React does *not* create a new div. It renders the children into `divNode`.
+  // `divNode` is any valid DOM node, regardless of its location in the DOM.
+  return React.createPortal(
+    this.props.children,
+    divNode,
+  );
+}
+```
+
+A typical use case for portals is when a parent component has an `overflow: hidden` or `z-index` style, but you need the child to visually "break out" of its container.
```

Comment:
Can we say explicitly somewhere that this is particularly useful for implementing modal dialogs, hovercards, and tooltips? Helps understand what it's for. Maybe worth noting the a11y requirements too though so people don't think it's too easy.